httputil/middleware.go

```diff
@@ -0,0 +1,15 @@
+package httputil
+
+import (
+	"log"
+	"net/http"
+)
+
+type LoggingMiddleware struct {
```

Comment:
I vote we keep the middleware out of this common repo for now. I'm not happy with it in it's current state. 
IMO our middleware system needs to be iterated on and handle more complex use-cases.